subprojects/tooling-api/src/integTest/groovy/org/gradle/integtests/tooling/r35/BuildProgressCrossVersionSpec.groovy

```diff
@@ -220,11 +220,21 @@ class BuildProgressCrossVersionSpec extends ToolingApiSpecification {
                     // Do nothing
                 }
             }
-            task runInWorker {
+            task runInProcess {
                 doLast {
                     def workerExecutor = gradle.services.get(WorkerExecutor)
                     workerExecutor.submit(TestRunnable) { config ->
-                        config.displayName = 'My Worker Action'
+                        config.forkMode = ForkMode.NEVER
+                        config.displayName = 'My in-process worker action'
```

Comment:
We need the display name before instantiating the runnable in order to create the corresponding build operations in the controlling build daemon.

But yes, the worker action should implement `Describable`, I'll do that change.